xdsl/interactive/app.py

```diff
@@ -347,6 +377,32 @@ def get_query_string(self) -> str:
         )
         return f"xdsl-opt -p {query}"
 
+    def update_number_of_input_ops(self, input_text: str) -> None:
+        if input_text == "":
+            self.input_number_of_ops.clear()
+        else:
+            ctx = MLContext(True)
+            for dialect in get_all_dialects():
+                ctx.load_dialect(dialect)
+            module = Parser(ctx, input_text).parse_module()
+            input_ops = count_number_of_operations(module)
+
+            self.input_number_of_ops.clear()
+            for k, v in input_ops.items():
+                self.input_number_of_ops.add_row(k, v)
```

Comment:
To me it feels like the op count data is state inherent to the app, and would benefit from being in a reactive property. Could you please add properties (probably tuples of key-value pairs) and watch methods to update the UI when the data changes?

Reply:
I think that the input_number_of_ops variable does not need to be a reactive property since the input text does not change frequently and when it does - it can just call the update_number_of_input ops function. 
I considered making the output_number_of_ops variable reactive (and adding watch/compute methods) but came to the conclusion that it was unnecessary.

Saw your comment in the other PR - is it okay if it stays just like that PR (except where I swap out dict for tuples) and not making those two variables reactive? what do u think? I think reactivity here would be overkill.

Reply:
I think it's worth doing, mostly because state that is explicit is better, that way we can easily introspect, debug, and extend it later, as opposed to being made of these temporary side-effecty things. Especially if you've already added it in the other PR, might as well bring it over here.

Reply:
Ah only saw later that you already made the change, thank you